endtoend/evaluators/metrics.go

```diff
@@ -0,0 +1,208 @@
+package evaluators
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	ptypes "github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	eth "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	e2e "github.com/prysmaticlabs/prysm/endtoend/params"
+	"github.com/prysmaticlabs/prysm/endtoend/types"
+	"github.com/prysmaticlabs/prysm/shared/p2putils"
+	"google.golang.org/grpc"
+)
+
+// MetricsCheck performs a check on metrics to make sure caches are functioning, and
+// overall health is good. Not checking the first epoch so the sample size isn't too small.
+var MetricsCheck = types.Evaluator{
+	Name:       "metrics_check_epoch_%d",
+	Policy:     afterNthEpoch(0),
+	Evaluation: metricsTest,
+}
+
+type equalityTest struct {
+	name  string
+	topic string
+	value int
+}
+
+type comparisonTest struct {
+	name               string
+	topic1             string
+	topic2             string
+	expectedComparison float64
+}
+
+var metricLessThanTests = []equalityTest{
+	{
+		name:  "memory usage",
+		topic: "go_memstats_alloc_bytes",
+		value: 100000000, // 100 Mb
```

Comment:
100MB only? I guess since it's only a few validators

Reply:
how do you get 100mb ? is there a better way to derive expected memory usage other than hardcoding it.

Reply:
I ran it and it seemed to be around 40mb normally. And yeah this is for only 64 validators so thats why its slow low.